Support dynamic favicons in the macOS browser

ayoy opened the pull request:
Task/Issue URL: https://app.asana.com/0/0/1209728950138483/f

### Description:
This change adds support for dynamic favicons, i.e. it handles changes to the favicon
made by the website asynchronously after load.
Old favicons user script was replaced with a content-isolated subfeature from C-S-S.
Favicons handling for a Tab was moved to a dedicated Tab Extension.
Minor updates to FaviconsManager were made to simplify the API.

### Testing Steps
See [this task](https://app.asana.com/0/72649045549333/1209718156754292) for the list of test websites and websites known to not work fine.
1. Open https://rpsthecoder.github.io/favicon-canvas-loader/ to see how dynamic favicons work.
2. Open https://privacy-test-pages.site/features/favicon/favicon_js/index.html for a more controllable example.
3. Click "Set default now”. Open History from Main Menu and verify that the favicon matches the history entry for this website.
4. Click “Set override now”. Open History from Main Menu and verify that the history entry displays an overridden favicon.
    * Note that you can test the same with History View (cmd+y) but it being a website causes some caching to happen which may present the outdated favicon until the tab is closed and reopened.
5. Repeat the same test (steps 3-4) with Bookmarks Manager and verify that it displays the most recently used favicon.

### Impact and Risks
Low: The only real change (not counting the tab extension) is to how frequently favicons may be reported by a website.

### Quality Considerations
As far as performance is concerned, I’ve tested https://rpsthecoder.github.io/favicon-canvas-loader/ on a 2015 MacBook Air with Big Sur where it worked perfectly fine, with no CPU spikes or beach-balling. More frequent updates than that are throttled on the C-S-S end (the native app won’t receive updates more frequently).

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

tomasstrba reviewed (approved):
This is great work, @ayoy 👏 

I like the improvement + encapsulation of the logic into the new tab extension.

1) I smoke tested a couple of popular websites, and favicons worked as expected ✔️ 
<img width="1352" alt="Screenshot 2025-03-21 at 09 28 52" src="https://github.com/user-attachments/assets/fd24bfae-87ac-4441-80dd-fc5e8f8532b8" />

2) https://rpsthecoder.github.io/favicon-canvas-loader/ works as expected ✔️
3) https://privacy-test-pages.site/features/favicon/favicon_js/index.html  ✔️ 
4) Regarding the history view, could we clear the cache before the view is reloaded to refresh favicons? (Not a blocker at all)
5) I've noticed favicons in the bookmarks bar are refreshed only after reopening of the window. It's not a blocker, but maybe there is a quick win to improve it?

tomasstrba commented:
One more thing I've noticed is the visual glitch I mentioned in the previous PR, but I've managed to reproduce it on the released build as well, so let's keep it out of the scope for this favicon improvement.

If you look at the tab favicon when switching the website, it switches back and forward between two websites for a short period of time.
https://github.com/user-attachments/assets/b5ed4839-14c7-48ff-a4de-8536f15d0393

ayoy commented:
Thanks for the review @tomasstrba!

Regarding open items:
4. We talked with Shane about force-refreshing favicons here and he had some ideas, but we wanted to handle them as a follow-up work in the history view. I added it to [Follow-up work for History View](https://app.asana.com/0/1201899738287924/1209694205095848/f).
5. I noticed the bookmarks bar too and it annoyed me. I could find an easy fix for that, and then I realized that Safari works the same so I skipped it for now 😅. I’ve added it to [Follow-up work for dynamic favicons](https://app.asana.com/0/1201899738287924/1209742255728000/f).
